File: 백준/Silver/11399. ATM/ATM.py
n = int(input())
arr = list(map(int, input().split()))

# 오름차순으로 숫자 배열 진행하기
arr.sort()

# 필요한 시간의 최솟값의 합 구하기
res = []
for i in range(n):
    sum_arr = sum(arr[0:i+1])
    res.append(sum_arr)

# 최종 값 출력
print(sum(res))

File: 백준/Silver/11399. ATM/ATM.java
import java.util.*;
import java.io.*;

public class Main{
    public static void main(String[] args) throws IOException {
       BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
       int n = Integer.parseInt(br.readLine());
       StringTokenizer st = new StringTokenizer(br.readLine());
       int[] arr = new int[n];
       for (int i = 0; i<n; i++) {
           int a = Integer.parseInt(st.nextToken());
           arr[i] = a;
       }
       Arrays.sort(arr);
       int sum = 0;
       for (int i=0; i<n; i++) {
           sum += arr[i] * (n-i);
       }
       System.out.println(sum);
    }
}